File: frontend/src/dto/places/PlaceResponseDto.ts
import type {PlaceType} from "../../enums/PlaceType.ts";
import type {ReviewResponseDto} from "../../dto/google/ReviewResponseDto.ts";

export interface PlaceResponseDto {
    id: number;
    placeType: PlaceType;
    name: string;
    description: string;
    rating: number;
    openingTime: string;
    closingTime: string;
    visitDuration: number;
    reviews: ReviewResponseDto[];
    latitude: number;
    longitude: number;
}

File: backend/src/main/java/com/example/backend/dto/activity/place/PlaceResponseDTO.java
package com.example.backend.dto.activity.place;

import com.example.backend.dto.review.ReviewResponseDTO;
import com.example.backend.model.enums.PlaceType;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class PlaceResponseDTO {

    Integer id;

    PlaceType placeType;

    String name;
    String description;
    double rating;

    String openingTime;
    String closingTime;
    int visitDuration;

    List<ReviewResponseDTO> reviews;

    Double latitude;
    Double longitude;
}



